configuration/configurator/ConfigSchemaHelper.cpp

```diff
@@ -0,0 +1,898 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jptree.hpp"
+#include "XMLTags.h"
+#include <cstring>
+#include "jfile.hpp"
+
+#include "ConfigSchemaHelper.hpp"
+#include "SchemaAttributes.hpp"
+#include "SchemaElement.hpp"
+#include "SchemaEnumeration.hpp"
+#include "ExceptionStrings.hpp"
+#include "BuildSet.hpp"
+#include "SchemaMapManager.hpp"
+#include "ConfigSchemaHelper.hpp"
+#include "ConfigFileUtils.hpp"
+#include "JSONMarkUp.hpp"
+
+
+#define SUCCESS 0
+#define FAILURE 1
+#define LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET \
```

Comment:
can this macro accept the length as a parameter? rather than being tied to a particular container?